Dockerfile.windows

```diff
@@ -118,30 +118,33 @@
 # The number of build steps below are explicitly minimised to improve performance.
 FROM microsoft/windowsservercore
 
+# Use PowerShell as the default shell
+SHELL ["powershell", "-command"]
+
 # Environment variable notes:
-#  - GO_VERSION must consistent with 'Dockerfile' used by Linux'.
+#  - GO_VERSION must consistent with 'Dockerfile' used by Linux.
 #  - FROM_DOCKERFILE is used for detection of building within a container.
 ENV GO_VERSION=1.7.3 `
     GIT_LOCATION=https://github.com/git-for-windows/git/releases/download/v2.10.1.windows.1/Git-2.10.1-64-bit.exe `
-    GOPATH=C:/go;C:/go/src/github.com/docker/docker/vendor `
+    GOPATH=C:\go;C:\go\src\github.com\docker\docker\vendor `
+    GOROOT=C:\go `
     FROM_DOCKERFILE=1
 
-WORKDIR c:/
+WORKDIR C:\
 
 RUN `
- setx /M Path "%Path%;c:\gcc\bin;c:\go\bin;" && `
- setx GOROOT "c:\go" && `
- powershell -command `
+  setx /M Path $($env:PATH+';C:\gcc\bin;C:\go\bin'); `
+  `
   $ErrorActionPreference = 'Stop'; `
   Function Download-File([string] $source, [string] $target) { `
-   $wc = New-Object net.webclient; $wc.Downloadfile($source, $target) `
+    $wc = New-Object net.webclient; $wc.Downloadfile($source, $target) `
   } `
   `
   Write-Host INFO: Downloading git...; `
-  Download-File %GIT_LOCATION% gitsetup.exe; `
+  Download-File $env:GIT_LOCATION gitsetup.exe; `
```

Comment:
Since this is PowerShell, I think the canonical casing is `Env`, but not very important